pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -50,6 +54,86 @@ def as_matrix(self, time):
     def check_transforms(self):
         self._A2B = check_transform(self._A2B)
         return self
+    
+
+class PandasTimeseriesTransform(TimeVaryingTransform):
+    """Transformation, which does change over time, represented in a pandas
+    dataframe.
+
+    Parameters
+    ----------
+    df : dataframe, shape (N, 7+)
+        Time-sequence of transformations, with each row representing a single
+        sample as position-quarternion (PQ) structure.
+        Refer to `pq_column_names()` for column naming.
+
+    """
+    def __init__(self, df):
+        self._df = df
+
+    @property
+    def _time_index(self):
+        return self._df.index.to_numpy()
+    
+    def _check_if_time_is_in_bounds(self, time):
+        time_index = self._time_index
+
+        if (time < time_index[0]) or (time > time_index[-1]):
+            raise ValueError("Time value is out of range!")
+
+    def as_matrix(self, time):
+        self._check_if_time_is_in_bounds(time)
```

Comment:
This is a bit to handle for this PR and most likely we don't want to handle this at all, but I was thinking about a case in which the time is out of range for this transformation, but then there is another path of transformations in the graph for which this time would be valid? We would have to handle that on a higher level.

Reply:
You are right, this check should be handled in the manager class. 

I was just thinking that if we we throw a (custom?) exception in some future, a higher level function could catch it and return a nice error message, e.g. something like "A -> B is not available at 42.42s"